modules/uhttp/http.go

```diff
@@ -29,122 +29,111 @@ import (
-
 	// default healthcheck endpoint
 	healthPath = "/health"
 
 	// default pprof endpoint
 	pprofPath = "/debug/pprof"
 )
 
-var _ service.Module = &Module{}
-
 // A Module is a module to handle HTTP requests
 type Module struct {
-	listener net.Listener
-	lock     sync.RWMutex
+	listener    net.Listener
+	lock        sync.RWMutex
+	l           *zap.Logger
+	handlerCtor fx.Component
 }
 
 // Config handles config for HTTP modules
 type Config struct {
-	Port    int           `yaml:"port"`
-	Timeout time.Duration `yaml:"timeout"`
+	Port    int           `yaml:"port" default:"3001"`
+	Timeout time.Duration `yaml:"timeout" default:"60s"`
 	Debug   bool          `yaml:"debug" default:"true"`
 }
 
-// GetHandlersFunc returns http handler created by caller
-type GetHandlersFunc func(service service.Host) http.Handler
-
 // New returns a new HTTP ModuleProvider.
-func New(handlerFunc GetHandlersFunc) service.ModuleProvider {
-	return service.ModuleProviderFromFunc("uhttp", func(host service.Host) (service.Module, error) {
-		handler := handlerFunc(host)
-		return newModule(host, handler)
-	})
-}
-
-func newModule(host service.Host, handler http.Handler) (*Module, error) {
-	// setup config defaults
-	cfg := Config{
-		Port:    defaultPort,
-		Timeout: defaultTimeout,
-	}
-	log := ulog.Logger(context.Background()).With(zap.String("module", host.ModuleName()))
-	if err := host.Config().Get("modules").Get(host.ModuleName()).Populate(&cfg); err != nil {
-		log.Error("Error loading http module configuration", zap.Error(err))
+func New(handlerCtor fx.Component) *Module {
+	if handlerCtor == nil {
+		panic("Handler constructor is nil")
 	}
 
-	serveMux := http.NewServeMux()
-	serveMux.Handle(healthPath, healthHandler{})
-
-	// TODO: pass in the auth client as part of module construction
-	authClient := auth.Load(host.Config(), host.Metrics())
-	stats := newStatsClient(host.Metrics())
-
-	handle :=
-		panicInbound(
-			metricsInbound(
-				tracingInbound(
-					authorizationInbound(handler, authClient, stats),
-				), stats,
-			), stats,
-		)
-	serveMux.Handle("/", handle)
-
-	if cfg.Debug {
-		serveMux.Handle(pprofPath, http.DefaultServeMux)
-	}
-	// Set up the socket
-	listener, err := net.Listen("tcp", fmt.Sprintf(":%d", cfg.Port))
-	if err != nil {
-		return nil, errors.Wrap(err, "unable to open TCP listener for HTTP module")
-	}
-
-	// finally, start the http server.
-	log.Info("Server listening on port", zap.Int("port", cfg.Port))
-
-	srv := &http.Server{
-		Handler: serveMux,
-	}
+	return &Module{handlerCtor: handlerCtor}
+}
 
-	go func() {
-		// TODO(pedge): what to do about error?
-		if err := srv.Serve(listener); err != nil {
-			log.Error("HTTP Serve error", zap.Error(err))
-		}
-	}()
-	return &Module{
-		listener: listener,
-	}, nil
+// Name returns uhttp.
+func (m *Module) Name() string {
+	return "uhttp"
 }
 
-// Start begins serving requests over HTTP
-func (m *Module) Start() error {
-	return nil
+type starter struct{}
```

Comment:
Why do you need empty starter struct? This is going to be a module construct, you can just return error.